src/main/java/frc/robot/FieldConstants.java

```diff
@@ -0,0 +1,174 @@
+        new ArrayList<>(13); // Starting at the right branch facing the driver station in clockwise
+
+    static {
+      // Initialize branch positions
+      for (int face = 0; face < 6; face++) {
+        Map<ReefHeight, Pose3d> fillRight = new HashMap<>();
+        Map<ReefHeight, Pose3d> fillLeft = new HashMap<>();
+        for (var level : ReefHeight.values()) {
+          Pose2d poseDirection = new Pose2d(CENTER, Rotation2d.fromDegrees(180 - (60 * face)));
+
+          fillRight.put(
+              level,
+              new Pose3d(
+                  new Translation3d(
+                      poseDirection
+                          .transformBy(
+                              new Transform2d(Offsets.ADJUST_X, Offsets.ADJUST_Y, new Rotation2d()))
+                          .getX(),
+                      poseDirection
+                          .transformBy(
+                              new Transform2d(Offsets.ADJUST_X, Offsets.ADJUST_Y, new Rotation2d()))
+                          .getY(),
+                      level.height),
+                  new Rotation3d(
+                      0,
+                      Units.degreesToRadians(level.pitch),
+                      poseDirection.getRotation().getRadians())));
+          fillLeft.put(
+              level,
+              new Pose3d(
+                  new Translation3d(
+                      poseDirection
+                          .transformBy(
+                              new Transform2d(
+                                  Offsets.ADJUST_X, -Offsets.ADJUST_Y, new Rotation2d()))
+                          .getX(),
+                      poseDirection
+                          .transformBy(
+                              new Transform2d(
+                                  Offsets.ADJUST_X, -Offsets.ADJUST_Y, new Rotation2d()))
+                          .getY(),
+                      level.height),
+                  new Rotation3d(
+                      0,
+                      Units.degreesToRadians(level.pitch),
+                      poseDirection.getRotation().getRadians())));
+        }
+        BRANCH_POSITIONS.add(fillLeft);
+        BRANCH_POSITIONS.add(fillRight);
+      }
+    }
+  }
+
+  public static class StagingPositions {
+    // Measured from the center of the ice cream
+    public static final Pose2d LEFT_ICE_CREAM =
+        new Pose2d(Units.inchesToMeters(48), Units.inchesToMeters(230.5), new Rotation2d());
+    public static final Pose2d MIDDLE_ICE_CREAM =
+        new Pose2d(Units.inchesToMeters(48), Units.inchesToMeters(158.5), new Rotation2d());
+    public static final Pose2d RIGHT_ICE_CREAM =
+        new Pose2d(Units.inchesToMeters(48), Units.inchesToMeters(86.5), new Rotation2d());
+  }
+
+  public enum Direction {
+    RIGHT(true),
+    LEFT(false);
+
+    Direction(boolean isRight) {
+      this.isRight = isRight;
+    }
+
+    public final boolean isRight;
+  }
+
+  public enum ReefHeight {
```

Comment:
```suggestion
  public enum ReefLevel {
```

It has height and pitch so we call it something else.